strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/ClassDependenciesGraph.java

```diff
@@ -56,26 +59,29 @@
     private static final Filter coreJava = new Filter(Collections.singletonList(""), Collections.singletonList("java.*"));
 
     private final JavaClassBuilder builder;
-    private final DirectedGraph<JavaElement, DefaultEdge> graph;
+    private final DirectedGraph<Element, DefaultEdge> graph;
     private final Filter filter;
     private final TestVerifier testVerifier;
     private final boolean enableTransitivity;
+    private final Path projectDir;
 
-    ClassDependenciesGraph(TestVerifier testVerifier, Configuration configuration) {
+    ClassDependenciesGraph(TestVerifier testVerifier, Configuration configuration, File projectDir) {
```

Comment:
This class is growing to do too many things:

- take care of graph construction, update and travesal
- finding resources / classes to be added to the graph
- transformation between types

Would it make sense to split it to several, more focused components?